thrift/src/main/java/com/linecorp/armeria/server/thrift/ThriftServiceSpecificationGenerator.java

```diff
@@ -105,8 +105,8 @@ public ServiceSpecification generate(Set<ServiceConfig> serviceConfigs,
                                     service.defaultSerializationFormat(),
                                     service.allowedSerializationFormats())));
 
-                    exampleHeaders.forEach((type, headersList) -> {
-                        if (serviceClass.isAssignableFrom(type)) {
+                    exampleHeaders.forEach((name, headersList) -> {
+                        if (serviceClass.getCanonicalName().equals(name)) {
```

Comment:
Yeah I have a fear of ```getName``` due to ```$``` but for thrift generated code it should always be the same.